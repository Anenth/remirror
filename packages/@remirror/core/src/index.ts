export * from './builtins';
export type { DelayedValue } from './commands';
export { delayedCommand, insertText, isDelayedValue, toggleMark } from './commands';
export * from './extension';
export type {
  AddCustomHandler,
  AddHandler,
  CustomHandlerMethod,
  HandlerKeyOptions,
} from './extension/base-class';
export type {
  AttributePropFunction,
  BaseFramework,
  CreateStateFromContent,
  FocusType,
  FrameworkOptions,
  FrameworkOutput,
  FrameworkProps,
  ListenerProps,
  PlaceholderConfig,
  RemirrorEventListener,
  RemirrorEventListenerProps,
  TriggerChangeProps,
  UpdateStateProps,
} from './framework';
export { Framework } from './framework';
export type { AnyRemirrorManager, CreateEditorStateProps, ManagerEvents } from './manager';
export { isRemirrorManager, RemirrorManager } from './manager';
export type {
  ApplyStateLifecycleProps,
  BaseExtensionOptions,
  ChangedOptions,
  CommandShape,
  CreateExtensionPlugin,
  DynamicOptionsOfConstructor,
  ExcludeOptions,
  ExtensionCommandFunction,
  ExtensionCommandReturn,
  ExtensionHelperReturn,
  ExtensionStore,
  GetChangeOptionsReturn,
  GetCommands,
  GetConstructor,
  GetHelpers,
  GetOptions,
  OnSetOptionsProps,
  OptionsOfConstructor,
  PickChanged,
  StateUpdateLifecycleProps,
  UpdateReason,
  UpdateReasonProps,
} from './types';
export * from '@remirror/core-constants';
export * from '@remirror/core-helpers';
export * from '@remirror/core-types';
export * from '@remirror/core-utils';
export type { CoreIcon } from '@remirror/icons';
