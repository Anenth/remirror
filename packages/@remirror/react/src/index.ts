export type { UseExtensionCallback, UseRemirrorProps, UseRemirrorReturn } from './hooks';
export {
  RemirrorContext,
  useActive,
  useCommands,
  useEditorDomRef,
  useEditorState,
  useEffectWithWarning,
  useExtension,
  useForceUpdate,
  useHelpers,
  useI18n,
  useManager,
  usePortalContainer,
  useRemirror,
  useRemirrorContext,
} from './hooks';
export { createReactManager } from './react-helpers';
export type { RemirrorProps } from './react-remirror';
export { EditorComponent, Remirror } from './react-remirror';
export type {
  CreateReactManagerOptions,
  DefaultReactExtensionUnion,
  GetRootPropsConfig,
  ReactExtensionUnion,
  ReactFrameworkOutput,
  RefKeyRootProps,
  RefProps,
  UseRemirrorContextType,
} from './react-types';
export * from './renderers';
export { createEditorView } from './ssr';
