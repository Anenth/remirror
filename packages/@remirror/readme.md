# @remirror

This folder contains all the scoped packages under the `@remirror/` npm organisation.
