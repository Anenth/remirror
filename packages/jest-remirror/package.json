{
  "name": "jest-remirror",
  "version": "0.13.1",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "ssr",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/next/packages/jest-remirror",
  "repository": "https://github.com/remirror/remirror/tree/master/packages/jest-remirror",
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "main": "dist/jest-remirror.cjs.js",
  "module": "dist/jest-remirror.esm.js",
  "types": "dist/jest-remirror.cjs.d.ts",
  "files": [
    "dist",
    "environment"
  ],
  "dependencies": {
    "@babel/runtime": "^7.10.1",
    "@remirror/core": "^0.11.0",
    "@remirror/pm": "^0.0.0",
    "@remirror/preset-core": "^0.0.0",
    "@remirror/react": "^0.13.1",
    "@remirror/react-utils": "^0.7.6",
    "@testing-library/react": "^10.0.4",
    "@types/node": "^14.0.5",
    "@types/react": "^16.9.35",
    "@types/react-dom": "^16.9.8",
    "@types/sanitize-html": "^1.23.0",
    "@types/testing-library__react": "^10.0.1",
    "jest-prosemirror": "^0.8.3",
    "jest-snapshot": "^26.0.1",
    "mutationobserver-shim": "^0.3.5",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "sanitize-html": "^1.24.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "preconstruct": {
    "entrypoints": [
      ".",
      "environment"
    ]
  }
}
