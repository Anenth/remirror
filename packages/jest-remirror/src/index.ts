export { ignoreJSDOMWarnings, jsdomExtras, jsdomPolyfill } from './jsdom-polyfills';
export { remirrorMatchers } from './jest-remirror-matchers';
export { renderEditor, RemirrorTestChain } from './jest-remirror-editor';
export { setupRemirrorEnvironment } from './jest-remirror-environment';
export type { RenderEditorParameter, TaggedProsemirrorNode } from './jest-remirror-types';

declare global {
  interface Window {
    hasWarnedAboutCancelAnimationFramePolyfill?: boolean;
    ignoreAllJSDOMWarnings: boolean;
  }
}
