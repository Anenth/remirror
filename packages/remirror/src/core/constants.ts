export * from '@remirror/core-constants';
