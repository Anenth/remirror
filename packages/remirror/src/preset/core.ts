export * from '@remirror/preset-core';
