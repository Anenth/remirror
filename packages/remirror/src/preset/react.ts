export * from '@remirror/preset-react';
