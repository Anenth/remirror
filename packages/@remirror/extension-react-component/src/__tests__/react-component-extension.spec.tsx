import { RemirrorTestChain } from 'jest-remirror';
import React, { ComponentType } from 'react';

import { ApplySchemaAttributes, NodeExtension, NodeExtensionSpec, NodeGroup } from '@remirror/core';
import { isExtensionValid } from '@remirror/testing';
import { act, createReactManager, RemirrorProvider, strictRender } from '@remirror/testing/react';

import { ReactComponentExtension } from '..';
import type { NodeViewComponentProps } from '../node-view-types';

test('`ReactComponentExtension`: is valid', () => {
  expect(isExtensionValid(ReactComponentExtension)).toBeTrue();
});

class TestExtension extends NodeExtension<{ useContent: boolean }> {
  createNodeSpec(extra: ApplySchemaAttributes): NodeExtensionSpec {
    return {
      attrs: {
        ...extra.defaults(),
        custom: { default: 'custom' },
      },
      content: 'inline*',
      group: NodeGroup.Block,
      toDOM: (node) => ['nav', extra.dom(node), 0],
    };
  }

  get name(): string {
    return 'test' as const;
  }

  ReactComponent: ComponentType<NodeViewComponentProps> = ({ node, forwardRef }) => {
    if (this.options.useContent) {
      return <p {...node.attrs} ref={forwardRef} />;
    }

    return <div>Ignore content</div>;
  };
}

test('NodeViews are created with content', () => {
  const chain = RemirrorTestChain.create(
    createReactManager([new ReactComponentExtension(), new TestExtension({ useContent: true })]),
  );

  strictRender(
    <RemirrorProvider manager={chain.manager}>
      <div />
    </RemirrorProvider>,
  );
  const {
    add,
    dom,
    nodes: { doc },
    attributeNodes: { test },
  } = chain;

  act(() => {
    add(doc(test({ custom: 'awesome' })('content<cursor>'))).insertText(' hello world');
  });

  expect(dom).toMatchSnapshot();
});

test('NodeViews are created without content', () => {
  const chain = RemirrorTestChain.create(
    createReactManager([new ReactComponentExtension(), new TestExtension({ useContent: false })]),
  );

  strictRender(
    <RemirrorProvider manager={chain.manager}>
      <div />
    </RemirrorProvider>,
  );

  const {
    add,
    dom,
    nodes: { doc },
    attributeNodes: { test },
  } = chain;

  act(() => {
    add(doc(test({ custom: 'awesome' })('content<cursor>'))).insertText('hello world');
  });

  expect(dom).toMatchSnapshot();
});
