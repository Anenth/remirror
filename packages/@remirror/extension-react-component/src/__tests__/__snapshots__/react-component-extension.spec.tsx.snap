// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeViews are created with content 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  aria-placeholder=""
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <nav>
    <p
      custom="awesome"
    >
      <span
        contenteditable="true"
      >
        content hello world
      </span>
    </p>
  </nav>
</div>
`;

exports[`NodeViews are created without content 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  aria-placeholder=""
  autofocus="false"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <nav>
    <div>
      Ignore content
    </div>
  </nav>
</div>
`;
